components/core/src/output.rs

```diff
@@ -99,65 +111,115 @@ impl<'a> StructuredOutput<'a> {
             content: content,
             verbose: None,
             color: None,
+            json: None,
         }
     }
 }
 
 // If we ever want to create multiple output formats in the future, we would do it here -
 // essentially create a flag we check to see what output you want, then call a different formatting
 // function. Viola!
+
 impl<'a> fmt::Display for StructuredOutput<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let verbose = self.verbose.unwrap_or(is_verbose());
-        let color = self.color.unwrap_or(is_color());
-        let preamble_color = if self.preamble == PROGRAM_NAME.as_str() {
-            Cyan
+        if self.json.unwrap_or(is_json()) {
+          jsonfmt(self, f)
+        }
+        else {
+          stdfmt(self, f)
+        }
+    }
+}
+
+fn stdfmt(output: &StructuredOutput, f: &mut fmt::Formatter) -> fmt::Result {
+    let verbose = output.verbose.unwrap_or(is_verbose());
+    let color = output.color.unwrap_or(is_color());
+
+    let re = Regex::new(r"\x1b\[[0-9;]*[a-zA-Z]").unwrap();
```

Comment:
The intention of this is to strip out any embedded ANSI color codes in output destined for a "no color" formatter, correct?

If so, is this because there are still places in Habitat's code that inappropriately embed color codes, or because output from user services may include these codes? If it's the former, we should make an issue to find those places and eliminate the color codes once and for all. In either case, we should document the intention behind this regex.

Additionally, we'll want to use the `lazy_static` macro to ensure this regex gets compiled only once, which will improve the performance of this, as it will be called _a lot_.

(Actually, we already have some functionality for this in another corner of the codebase! See https://github.com/habitat-sh/habitat/blob/db42e8970682f0a803649b0bbda9483928c70a21/components/builder-protocol/src/jobsrv.rs#L203-L218)

Reply:
This is the intention, yes.

The Habitat colour thing is an odd issue - I had actually chased them down to remove but then you don't get them with the default behavior (highlighted words etc..). I think this is nice for console output. As there is a single field for Habitat and service outputs (it could be one or the other) it becomes difficult.

The other option would be to make it so there was a 'habitat' created output string, which was separate from the service created output string.

I was anticipating some discussion on if we should even be stripping codes from service output, I think yes (or don't use the option) but I'm sure a lot of people would say no (I would argue if you don't want it don't use it).